### PR TITLE
Recover RPA Jastrow 

### PR DESCRIPTION
**Pull request by rcclay**
This recovers the old RPA Jastrow code.  Achieved by 
1.) implementing evalGrad, ratioGrad, and evaluateLog for compatibility with recent releases (in recent memory).  
2.) Adding somewhat redundant code to handle default "USE_REAL_STRUCT_FACTOR" use case.

This passes finite-diff and ratio tests.  This is an initial staging for a more generalized RPA jastrow.

### CODE REVIEW

File: src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp [hunk 1]
```diff
@@ -92,12 +99,21 @@ LRTwoBodyJastrow::evaluateLog(ParticleSet& P,
 {
   RealType sum(0.0);
 #if defined(USE_REAL_STRUCT_FACTOR)
-  APP_ABORT("LRTwoBodyJastrow::evaluateLog ");
+ // APP_ABORT("LRTwoBodyJastrow::evaluateLog ");
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp]: Remove commented-out line

File: src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp [hunk 2]
```diff
@@ -132,32 +166,8 @@ LRTwoBodyJastrow::evaluateLog(ParticleSet& P,
...
-//          l += ksq[ki]*(Fk[ki]-skp.real());
-//#endif
-//        }
-//        sum+=(U[iat]=res);
-//        G[iat]+=(dU[iat]=g);
-//        L[iat]+=(d2U[iat]=l);
-//      }
-//
-//      return sum*0.5;
+  //return sum;
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp]: remove line

File: src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp [hunk 3]
```diff
@@ -170,7 +180,34 @@ LRTwoBodyJastrow::ratio(ParticleSet& P, int iat)
   //restore, if called should do nothing
   NeedToRestore=false;
 #if defined(USE_REAL_STRUCT_FACTOR)
-  APP_ABORT("LRTwoBodyJastrow::ratio(ParticleSet& P, int iat)");
+  //APP_ABORT("LRTwoBodyJastrow::ratio(ParticleSet& P, int iat)");
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp]: Remove line.

File: src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp [hunk 4]
```diff
@@ -386,12 +605,19 @@ LRTwoBodyJastrow::updateBuffer(ParticleSet& P, PooledData<RealType>& buf,
 {
   LogValue=evaluateLog(P,P.G,P.L);
 #if defined(USE_REAL_STRUCT_FACTOR)
-  APP_ABORT("LRTwoBodyJastrow::updateBuffer");
+//  APP_ABORT("LRTwoBodyJastrow::updateBuffer");
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp]: Remove line

File: src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp [hunk 5]
```diff
@@ -400,22 +626,37 @@ LRTwoBodyJastrow::updateBuffer(ParticleSet& P, PooledData<RealType>& buf,
...
+  buf.get(Rhok_i.first_address(), Rhok_i.last_address());
+#else
   buf.get(Rhok.first_address(), Rhok.last_address());
+#endif
   buf.get(U.first_address(), U.last_address());
   buf.get(d2U.first_address(), d2U.last_address());
   buf.get(FirstAddressOfdU,LastAddressOfdU);
 #if defined(USE_REAL_STRUCT_FACTOR)
-  APP_ABORT("LRTwoBodyJastrow::copyFromBuffer");
+//  APP_ABORT("LRTwoBodyJastrow::copyFromBuffer");
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/LRTwoBodyJastrow.cpp]: remove line

File: src/QMCWaveFunctions/Jastrow/RPAJastrow.cpp [hunk 6]
```diff
@@ -67,6 +68,7 @@ bool RPAJastrow::put(xmlNodePtr cur)
   params.add(Rs,"rs","double");
   params.add(Kc,"kc","double");
   params.put(cur);
+  app_log()<<"!!!KC = "<<Kc<<"\n";
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/RPAJastrow.cpp]: Remove this or expand the output message.

File: src/QMCWaveFunctions/Jastrow/RPAJastrow.cpp [hunk 7]
```diff
@@ -285,8 +306,15 @@ RPAJastrow::evaluateLog(ParticleSet& P,
                         ParticleSet::ParticleGradient_t& G, ParticleSet::ParticleLaplacian_t& L)
 {
   LogValue=0.0;
+//  app_log()<<"RPAJastrow.  Calling evaluateLog!\n";
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/RPAJastrow.cpp]: Remove commented-out lines (here and below)

File: src/QMCWaveFunctions/Jastrow/RPAJastrow.h [hunk 8]
```diff
@@ -34,8 +35,9 @@ namespace qmcplusplus
 struct RPAJastrow: public OrbitalBase
 {
   typedef LRHandlerBase HandlerType;
-  typedef CubicBspline<RealType,LINEAR_1DGRID,FIRSTDERIV_CONSTRAINTS> SplineEngineType;
-  typedef CubicSplineSingle<RealType,SplineEngineType> FuncType;
+  typedef BsplineFunctor<RealType> FuncType;
+//  typedef CubicBspline<RealType,LINEAR_1DGRID,FIRSTDERIV_CONSTRAINTS> SplineEngineType;
```
Comment (markdewing) [inline src/QMCWaveFunctions/Jastrow/RPAJastrow.h]: Remove these lines, or add a comment if there's a reason to leave them here.

**Comment by markdewing** [comment]
Could you make these changes in a branch and submit a new pull request from the branch?  That should remove the 'sync with main' commits.

**Comment by rcclay** [comment]
@markdewing  Thanks for revisions.  I'll resubmit them from my kspacejastrow branch. 

@prckent My tests were originally done on an Li2O unit cell, but I can tack the wavefunction block in any system you think is best.  Caveat:  I'm not comfortable yet with using this as a production level jastrow--while the code runs and passes the wavefunction tests, I don't yet know if the code is correct on a high level.  Even if it's not though, the current state should be good at detecting breakages.  I'll write a warning/disclaimer banner to standard out.

**Pull request closed without merging**